python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -244,6 +248,46 @@ def get_partitions(
 
         return partitions
 
+    def _get_validated_time_window_for_partition_key(
+        self, partition_key: str, current_time: Optional[datetime] = None
+    ) -> Optional[TimeWindow]:
+        """Returns a TimeWindow for the given partition key if it is valid, otherwise returns None.
+        """
+        try:
```

Comment:
Could we invoke `has_partition_key(...)` here instead of duplicating the implementation?

Reply:
We could. I originally structured it this way so we wouldn't have to fetch the time window twice. If we called has_partition_key, we'd still then have to refetch the time window if has_partition_key returned true

Reply:
would it make sense to factor our the implementation to a private function that's used by both?

Reply:
Ohhh I misunderstood